src/openfermion/measurements/_probability_distributions.py

```diff
@@ -0,0 +1,379 @@
+        """
+        variances = []
+
+        if vectors is None:
+            vectors = self._fourier_vectors
+
+        for j in range(self._num_vectors):
+            if (vectors[2, j]**2 + vectors[1, j]**2) == 0:
+
+                variances.append(maxvar)
+
+            else:
+                variances.append(min(
+                    maxvar,
+                    4 / (vectors[2, j]**2+vectors[1, j]**2) - 1))
+        return variances
+
+    def _make_dist(self, vector, angle):
+        """
+        converts a frequency vector into a function evaluated at theta
+        Args:
+            vector (numpy array): a fourier vector P(phi)
+            angle (float): the angle to evaluate P at
+
+        Returns:
+            (float) P(angle)
+        """
+        res = vector[0]
+        for j in range(1, self._num_freqs):
+            res += vector[2*j-1] * sin(j*angle)
+            res += vector[2*j] * cos(j*angle)
+        return res / (2*pi)
+
+    def _mlikelihood(self, a_vec):
+        # Calculates the negative likelihood of a set of amplitudes
+        # generating the observed measurements
+        return -sum([numpy.log(p_vec.dot(a_vec[:len(p_vec)]))
+                     for p_vec in self._p_vecs]) +\
+            self._init_mlikelihood(a_vec)
+
+    def _diff_mlikelihood(self, a_vec):
+        # The derivative of the above function
+        return -sum([p_vec/(p_vec.dot(a_vec[:len(p_vec)]))
+                     for p_vec in self._p_vecs]) +\
+            self._dinit_mlikelihood(a_vec)
+
+    def _single_diff(self, p_vec, a_vec):
+        # The first derivative of a single term in the
+        # likelihood function.
+        return -p_vec / numpy.dot(p_vec, a_vec[:len(p_vec)])
+
+    def _jacobian_term(self, p_vec, a_vec):
+        # The second derivative of the above function,
+        # evaluated for a single p vector - to evaluate
+        # and store
+        return numpy.dot(p_vec[:, numpy.newaxis], p_vec[numpy.newaxis, :]) /\
+            numpy.dot(p_vec, a_vec[:len(p_vec)]) ** 2
+
+    def _calculate_jacobian(self):
+        # Recalculates the Jacobian based on all previously
+        # calculated p_vecs.
+        self._jacobian = sum(
+            [self._jacobian_term(
+                p_vec,
+                self._amplitude_estimates)
+             for p_vec in self._p_vecs]) + self._jinit_mlikelihood()
+
+    def _init_mlikelihood(self, a_vec):
+        # A normal distributed initial guess at the amplitudes
+        return numpy.dot(
+            0.5*(a_vec-self._prior_amplitude_mean)[
+                numpy.newaxis, :],
+            numpy.dot(self._inv_covar_mat,
+                      (a_vec-self._prior_amplitude_mean)[
+                        :, numpy.newaxis])).item()
+
+    def _dinit_mlikelihood(self, a_vec):
+        # The derivative of the above likelihood function
+        return numpy.dot(
+            self._inv_covar_mat,
+            (a_vec-self._prior_amplitude_mean))
+
+    def _jinit_mlikelihood(self):
```

Comment:
None of these hidden methods are being used in any non-hidden methods in this class. So you should unhide them or get rid of them.

Reply:
Thanks for noticing this, I'll unhide them (they're used by BayesEstimator).